core/src/main/java/com/datastax/oss/driver/internal/core/config/typesafe/DefaultDriverConfigLoader.java

```diff
@@ -58,8 +60,10 @@
         ConfigFactory.invalidateCaches();
         // The thread's context class loader will be used for application classpath resources,
         // while the driver class loader will be used for reference classpath resources.
-        return ConfigFactory.defaultApplication()
```

Comment:
We have also a similar pattern in the `DefaultProgrammaticDriverConfigLoaderBuilder`:
```
  public static final Supplier<Config> DEFAULT_FALLBACK_SUPPLIER =
      () ->
          ConfigFactory.defaultApplication()
              // Do not remove root path here, it must be done after merging configs
              .withFallback(ConfigFactory.defaultReference(CqlSession.class.getClassLoader()));
```
and 
```
  public DefaultProgrammaticDriverConfigLoaderBuilder(@NonNull ClassLoader appClassLoader) {
    this(
        () ->
            ConfigFactory.defaultApplication(appClassLoader)
                .withFallback(ConfigFactory.defaultReference(CqlSession.class.getClassLoader())),
        DefaultDriverConfigLoader.DEFAULT_ROOT_PATH);
  }
```
I am not sure if progrmatticConfigLoader should also accept `ConfigFactory.defaultOverrides()`, wdyt?

Reply:
No it should not change, the default overrides are applied later on, see [here](https://github.com/datastax/java-driver/blob/1427be868bcbb115258e87cccb1cffcf54c82e04/core/src/main/java/com/datastax/oss/driver/internal/core/config/typesafe/DefaultProgrammaticDriverConfigLoaderBuilder.java#L250).